cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +70,33 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def _find_reachable_deps(yarn_lock, dep):
```

Comment:
Just pointing out that, since we're restarting the BFS for every non-dev top level dependency, we revisting every transitive dependency that is common to more than one top level dependency. Like:
```
A---- 
     |
     |----C   
     |
B----
``` 
C (and its dependency tree) will be visited twice in this case.
I don't think this is a big deal though, since the time for this algorithm to run is tiny fraction of all the time Cachito takes to process a request. :)

Reply:
I did think this as well. Would memoization save noticeable time for most Cachito requests?

Reply:
Solved!